packages/oui-file/README.md

```diff
@@ -0,0 +1,117 @@
+    <oui-button type="submit">Submit</oui-button>
+</form>
+```
+
+## Multiple files
+
+### Basic
+
+```html:preview
+<oui-file model="$ctrl.modelMultiple" maxsize="150000" multiple></oui-file>
+```
+
+### With preview
+
+<oui-message type="info" dismissable="false">Preview works only with `image/*` files.</oui-message>
+
+```html:preview
+<oui-file model="$ctrl.modelMultiplePreview" maxsize="150000" multiple preview></oui-file>
+```
+
+## Drag & Drop area
+
+### Basic
+
+```html:preview
+<oui-file model="$ctrl.modelDroparea" maxsize="150000" droparea></oui-file>
+```
+
+### With preview
+
+<oui-message type="info" dismissable="false">Preview works only with `image/*` files.</oui-message>
+
+```html:preview
+<oui-file model="$ctrl.modelDropareaPreview" maxsize="150000" droparea preview></oui-file>
+```
+
+## API
+
+| Attribute         | Type              | Binding   | One-time binding  | Values            | Default   | Description
+| ----              | ----              | ----      | ----              | ----              | ----      | ----
+| `model`           | array&lt;file&gt; | =         | no                | n/a               | n/a       | model bound to component
+| `id`              | string            | @?        | yes               | n/a               | n/a       | id attribute of form input
+| `name`            | string            | @?        | yes               | n/a               | n/a       | name attribute of form input
+| `placeholder`     | string            | @?        | yes               | n/a               | n/a       | placeholder text
+| `accept`          | string            | @?        | yes               | n/a               | n/a       | accept attribute of file input
+| `maxsize`         | number            | <?        | no                | n/a               | n/a       | maximum file size in byte
+| `disabled`        | boolean           | <?        | no                | `true`, `false`   | `false`   | disabled flag
+| `required`        | boolean           | <?        | no                | `true`, `false`   | `false`   | required flag
+| `multiple`        | boolean           | <?        | yes               | `true`, `false`   | `false`   | multiple flag
+| `droparea`        | boolean           | <?        | yes               | `true`, `false`   | `false`   | droparea flag
+| `preview`         | boolean           | <?        | yes               | `true`, `false`   | `false`   | preview flag
+| `on-select`       | function          | &         | no                | n/a               | n/a       | handler triggered when files are selected
+
+## Configuration
+
+The file component can be globally configured with a provider.
+
+```js
+angular.module("myModule", [
+    "oui.file"
+]).config(ouiFileConfigurationProvider => {
+    ouiFileConfigurationProvider.setTranslations({ // default translations
+        attachmentsHeading: "Attachment(s)",
+        dropArea: "Attach document(s) by drap and drop or",
```

Comment:
The `s` could be deduced from the presence or not of the `multiple` attribute :)

Reply:
Drop area is multiple only